qemu/tests/blockdev_commit_specify_node.py

```diff
@@ -0,0 +1,60 @@
+import logging
+
+from virttest.qemu_monitor import QMPCmdError
+
+from provider import backup_utils
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitSpecifyNode(BlockDevCommitTest):
+
+    def commit_snapshots(self):
+        device = self.params.get("device_tag")
+        device_params = self.params.object_params(device)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        options = ["base-node", "top-node", "speed"]
+        arguments = self.params.copy_from_keys(options)
+        test_scenario = self.params["test_scenario"]
+        if test_scenario == 'base_same_with_top':
+            arguments["base-node"] = self.get_node_name(snapshot_tags[-2])
+            arguments["top-node"] = self.get_node_name(snapshot_tags[-2])
+            device = self.get_node_name(snapshot_tags[-1])
+        if test_scenario == 'parent_as_top_child_as_base':
+            arguments["base-node"] = self.get_node_name(snapshot_tags[-2])
+            arguments["top-node"] = self.get_node_name(snapshot_tags[0])
+            device = self.get_node_name(snapshot_tags[-1])
+        commit_cmd = backup_utils.block_commit_qmp_cmd
+        cmd, args = commit_cmd(device, **arguments)
+        try:
+            self.main_vm.monitor.cmd(cmd, args)
+        except QMPCmdError as e:
+            logging.info("Error message is %s", e.data)
+            if self.params.get("error_msg") not in str(e.data):
```

Comment:
For error_msg is required, and always set in cfg, here you can use self.params['error_msg'] instead of a get() method, fyi.